src/mlpack/methods/reinforcement_learning/environment/mountain_car.hpp

```diff
@@ -187,7 +187,7 @@ class MountainCar
     State state;
     stepsPerformed = 0;
     state.Velocity() = 0.0;
-    state.Position() = arma::as_scalar(arma::randu(1)) * 0.2 - 0.6;
```

Comment:
It might be safer to use `Random()` from `core/math/random.hpp` instead.  I'm not sure if that's the cause of the `as_scalar` exception you're seeing.

Reply:
I removed this one and it was still causing the problem, I used directly `arma::randu()`

Reply:
What else did you try to debug the issue?  I usually will use `gdb` to figure out where the exception is being thrown, and then from there generally the problem is pretty easy to track down. :+1:

Reply:
Technically, it is all related to the change of `Imbue`. 
When initializing the matrix with `arma::randn<MatType>(n_rows, n_cols, distr_param(stddev));` I got the `as_scalar` problem in the greedy policy in reinforcement learning code base.
if I change the matrix initialization to `W = arma::randn<MatType>(n_rows, n_cols) * std::sqrt(variance);` the SAC `PendilumWithSAC` test goes into an infinite loop.

Reply:
For me all of these terms are exactly the same, yet our tests are not behaving the same way, so either there is a problem in the tests or there is a bug somewhere that already exists or I am adding it with out being aware.